src/Tooltip/index.js

```diff
@@ -459,40 +471,42 @@ class Tooltip extends React.PureComponent<Props, State> {
         >
           {children}
         </StyledTooltipChildren>
-        <Portal element="tooltips">
-          <StyledTooltip role="tooltip" data-test={dataTest}>
-            <StyledTooltipOverlay
-              onClick={this.handleClose}
-              shownMobile={shownMobile}
-              ref={this.overlay}
-            />
-            <StyledTooltipWrapper
-              shown={shown}
-              shownMobile={shownMobile}
-              position={position}
-              align={align}
-              size={size}
-              ref={this.tooltip}
-              onMouseEnter={this.handleIn}
-              onClick={this.handleClose}
-              onMouseLeave={this.handleOut}
-              containerTop={containerTop}
-              containerLeft={containerLeft}
-              containerHeight={containerHeight}
-              containerWidth={containerWidth}
-              tooltipHeight={tooltipHeight}
-              tooltipWidth={tooltipWidth}
-              contentHeight={contentHeight}
-            >
-              <StyledTooltipContent ref={this.content}>{content}</StyledTooltipContent>
-              <StyledTooltipClose>
-                <Button type="secondary" block onClick={this.handleClose} ref={this.closeButton}>
-                  {closeText}
-                </Button>
-              </StyledTooltipClose>
-            </StyledTooltipWrapper>
-          </StyledTooltip>
-        </Portal>
+        {(shown || shownMobile) && (
```

Comment:
I would add another state that handles rendering into DOM and the `shown` and `showMobile` would be used only for visually displaying it after all calculations etc. is ready.